### PR TITLE
Feature/fix jvm compatibility issue

### PR DESCRIPTION
**Pull request by Vikrech**
### Purpose 

- This pull request aims to fix the issue `Compilation error due to mismatch between compileDebugJavaWithJavac and compileDebugKotlin`.

### Description

- On the project when we migrated android part to use Plugin DSL, we faced this issue: "Inconsistent JVM-target compatibility detected for tasks 'compileDebugJavaWithJavac' (1.8) and 'compileDebugKotlin' (17)". After analysing the Gradle script output, we discovered that there might be some missing configuration for the plugin.

Screenshot 2024-11-01 at 11 29 35 

### Changes:
- Added `compileOptions` and `kotlinOptions` bloc's to `android/build.gradle` with jvmTarget 1.8
- Replaced `jcenter` with `mavenCentral`
- Updated kotlin to version `1.8.10`
- Updated gradle dependency version to 7.4.2
- Fixed issue in example;

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -2,22 +2,22 @@ group 'com.logger.logging_to_logcat'
...
+    ext.kotlin_version = '1.8.10'
     repositories {
         google()
-        jcenter()
+        mavenCentral()
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:4.1.0'
+        classpath 'com.android.tools.build:gradle:7.4.2'
```
Comment (jogboms) [inline android/build.gradle:12]: Maybe 8+?

Comment (Vikrech) [reply android/build.gradle]: The version of gradle has been updated to 8.4.0 [005c6e7](https://github.com/ChristianKleineidam/logging_to_logcat/pull/6/commits/005c6e7997248c5c1e0ebde5f091de3c4af07c14)

**Comment by Vikrech** [comment]
Here is an example code running on the Google Pixel 4 with the changes from this pull request. 

https://github.com/user-attachments/assets/71bdefe3-be74-43cb-ac60-75f17c59c027


The changes in current pull request should completely fix this issue https://github.com/ChristianKleineidam/logging_to_logcat/issues/4


In case if it is matter the output from the: `flutter doctor -v`

```
[✓] Flutter (Channel stable, 3.24.3, on macOS 14.7 23H124 darwin-arm64, locale en-UA)
    • Flutter version 3.24.3 on channel stable at /Users/Viktor_Krechetov/fvm/versions/3.24.3
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2663184aa7 (7 weeks ago), 2024-09-11 16:27:48 -0500
    • Engine revision 36335019a8
    • Dart version 3.5.3
    • DevTools version 2.37.3

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/Viktor_Krechetov/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/Viktor_Krechetov/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.10+0-17.0.10b1087.21-11572160)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15F31d
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2023.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.10+0-17.0.10b1087.21-11572160)

[✓] IntelliJ IDEA Community Edition (version 2024.2.4)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 82.0.3
    • Dart plugin version 242.22855.32

[✓] VS Code (version 1.94.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.98.0

[✓] Connected device (6 available)            
    • Pixel 4 (mobile)                 • 99261FFAZ005ES            • android-arm64  • Android 13 (API 33)
    • sdk gphone64 arm64 (mobile)      • emulator-5554             • android-arm64  • Android 13 (API 33) (emulator)
    • Viktor’s iPhone 13 mini (mobile) • 00008110-0019556C0192801E • ios            • iOS 17.7 21H16
    • macOS (desktop)                  • macos                     • darwin-arm64   • macOS 14.7 23H124 darwin-arm64
    • Mac Designed for iPad (desktop)  • mac-designed-for-ipad     • darwin         • macOS 14.7 23H124 darwin-arm64
    • Chrome (web)                     • chrome                    • web-javascript • Google Chrome 130.0.6723.92

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

File: pubspec.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 name: logging_to_logcat
 description: Adds activateLogcat() method to logging's Logger class which results in everything that's
   logged to Logger being displayed by Android's logcat.
-version: 1.0.3
+version: 1.0.4
```
Comment (jogboms) [inline pubspec.yaml:4]: Should this instead be a minor bump?

**Pull request merged**